### PR TITLE
Add on error handler for snap blocks

### PR DESCRIPTION
**Pull request by asoto-iov**
The main purpose of this PR is to add the onStateChunkResponseError method, so if during the process of a stateChunkResponse, an error happens we ask for the same stateChunk again and to another peer if possible.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

**Comment by rmoreliovlabs** [comment]
The changes look good from what I’ve seen so far—nice work! To help me review everything thoroughly, could you add a bit more context? Specifically:

- A brief overview of what’s been changed and why.
- Implementation Details: How the new error handling works in practice, including how it integrates with the `SnapshotProcessor` class and the `processStateChunkResponse` and `processSnapStatusResponse` method

This extra info will help me get a clearer picture of how everything fits together and make the review process smoother, appreciate your help with this!

File: rskj-core/src/main/java/co/rsk/net/sync/BlockConnectorHelper.java [hunk 1]
```diff
@@ -30,37 +30,41 @@
...
-        logger.info("Start connecting Blocks");
+    public void startConnecting(List<Pair<Block, BlockDifficulty>> blockAndDifficultiesList) {
         if (blockAndDifficultiesList.isEmpty()) {
             logger.debug("Block list is empty, nothing to connect");
             return;
         }
+
+        blockAndDifficultiesList.sort(new BlockAndDiffComparator());
+        Block child = null;
+        logger.info("Start connecting Blocks. To connect from {} to {} - Total: {}",
```
Comment (nagarev) [inline rskj-core/src/main/java/co/rsk/net/sync/BlockConnectorHelper.java:46]:
```suggestion
        logger.info("Start connecting blocks ranging from {} to {} - Total: {}",
```

File: rskj-core/src/main/java/co/rsk/net/sync/SnapSyncState.java [hunk 2]
```diff
@@ -41,7 +41,7 @@
 
 public class SnapSyncState extends BaseSyncState {
 
-    private static final Logger logger = LoggerFactory.getLogger("syncprocessor");
+    private static final Logger logger = LoggerFactory.getLogger("snapSyncState");
```
Comment (nagarev) [inline rskj-core/src/main/java/co/rsk/net/sync/SnapSyncState.java:44]:
```suggestion
    private static final Logger logger = LoggerFactory.getLogger("SnapSyncState");
```

File: rskj-core/src/test/java/co/rsk/net/sync/SnapSyncStateTest.java [hunk 3]
```diff
@@ -219,10 +225,86 @@ void givenOnSnapStateChunkIsCalled_thenJobIsAddedAndRun() throws InterruptedExce
         assertEquals(msg, jobArg.getValue().getMsg());
     }
```
Comment (fmacleal) [inline rskj-core/src/test/java/co/rsk/net/sync/SnapSyncStateTest.java:227]: question:

What's the benefit of test the set and get? Do they have something different than just set the variable?

File: rskj-core/src/test/java/co/rsk/net/SnapshotProcessorTest.java [hunk 4]
```diff
@@ -333,6 +334,35 @@ void processStateChunkRequestInternal(Peer sender, SnapStateChunkRequestMessage
         assertEquals(msg, jobArg.getValue().getMsg());
     }
 
+    @Test
```
Comment (fmacleal) [inline rskj-core/src/test/java/co/rsk/net/SnapshotProcessorTest.java:337]: suggestion: 

Is it possible to replicate this error without so many coupled mocks? Isn't possible to execute the test like the others one and just make the `peersInformation` throws an exception to trigger the method you want? Something like:

`doThrow(new RuntimeException("Some error")).when(peersInformation.getOrRegisterPeer(any()));`

**Review by fmacleal — COMMENTED** [review]
Good job! I just have a few comments.

File: rskj-core/src/test/java/co/rsk/net/sync/SnapSyncStateTest.java (hunk 3 above)
Comment (fmacleal) [reply rskj-core/src/test/java/co/rsk/net/sync/SnapSyncStateTest.java:227]: We can improve that in a future ticket. I am closing the threads now.

File: rskj-core/src/test/java/co/rsk/net/SnapshotProcessorTest.java (hunk 4 above)
Comment (fmacleal) [reply rskj-core/src/test/java/co/rsk/net/SnapshotProcessorTest.java:337]: We can improve that in a future ticket. I am closing the conversation now.

**Pull request merged**